core/Cargo.lock

```diff
@@ -2770,6 +2860,12 @@ version = "2.0.0"
 source = "registry+https://github.com/rust-lang/crates.io-index"
 checksum = "e6d5a32815ae3f33302d95fdcb2ce17862f8c65363dcfd29360480ba1001fc9c"
 
+[[package]]
+name = "futures"
+version = "0.1.31"
+source = "registry+https://github.com/rust-lang/crates.io-index"
+checksum = "3a471a38ef8ed83cd6e40aa59c1ffe17db6855c18e3604d9c4ed8c08ebc28678"
```

Comment:
It's `futures-util`, `fbthrift-*` used in `rust-nebula` and `bb8` depends on `futures` and `futures` depends on `futures-util`. [futures-utils/dependencies](https://crates.io/crates/futures-util/0.3.30/dependencies) displayed that `futures=0.1.25`.

`cargo tree` is as follows:
```
├── bb8 v0.8.5
│   ├── async-trait v0.1.82 (proc-macro) (*)
│   ├── futures-util v0.3.30
│   │   ├── futures v0.1.31
│   │   ├── futures-channel v0.3.30
│   │   │   ├── futures-core v0.3.30
│   │   │   └── futures-sink v0.3.30
```

Does it need to be removed? I just don't know how to do it:(

Reply:
Seems someone has enabled `futures-util`'s `compat` feature in wrong.

---

Ok, I got it: https://github.com/bk-rs/fbthrift-git-rs/blob/36ddca22a4aca8bbf02482ca20ff3fe525a24a87/Cargo.toml#L17

Reply:
vkill has contributed almost all of fbThrift's packages, can we remove this without communicating with him?

Reply:
I'm not sure. Can you contact @vkill? Or can we obtain write permission for the `bk-rs` organization? Maybe @wey-gu can help a bit.

Reply:
Perhaps it's possible, but I don't have high hopes. If I can establish a stable connection with vkill, I won't have to release a new `rust-nebula` crate. I will make modifications directly on his `nebula-rust` crate:(

Reply:
I briefly reviewed all the fbthrift related code that `rust-nebula` depends on and found that there is nothing that requires the use of `compat`. Therefore, I temporarily moved all these code to `rust-nebula`, removed `compat`, and made the necessary changes.

Now `rust-nebula` does not rely on `futures v0.1`, and it has also passed all tests (as above, I edited it).

I will also continue to communicate with vkill and Facebook, and if they are willing to make changes, I will update my code. If they don't want to, I may release these fbthrift package separately after `rust-nebula` becomes more mature (because there are 8 crates to be changed, so I don't really want to publish them directly to occupy so many names now), but this has no impact on OpenDAL.

Reply:
Thank you for your effort, that's really great.